flutter_local_notifications/ios/Classes/FlutterLocalNotificationsPlugin.m

```diff
@@ -234,6 +253,100 @@ - (void)pendingNotificationRequests:(FlutterResult _Nonnull)result {
   }
 }
 
+- (UNNotificationCategoryOptions)parseNotificationCategoryOptions:
+    (NSArray *)options API_AVAILABLE(ios(10.0)) {
+  int result = UNNotificationCategoryOptionNone;
+
+  for (NSNumber *option in options) {
+    result |= [option intValue];
+  }
+
+  return result;
+}
+
+- (UNNotificationActionOptions)parseNotificationActionOptions:(NSArray *)options
+    API_AVAILABLE(ios(10.0)) {
+  int result = UNNotificationActionOptionNone;
+
+  for (NSNumber *option in options) {
+    result |= [option intValue];
+  }
+
+  return result;
+}
+
+/// Extracts notification categories from [arguments] and configures them as
+/// appropriate.
+///
+/// This code will simply return the `completionHandler` if not running on a
+/// compatible iOS version or when no categories were specified in [arguments].
+- (void)configureNotificationCategories:(NSDictionary *_Nonnull)arguments
+                                 result:(FlutterResult _Nonnull)result {
```

Comment:
Was there a reason to include the `result` object here? From what I could tell, it shouldn't be needed here and using here could lead to issues as calling `result(...)` signals to Flutter that the `Future` related to `initialize()` method has completed. This is actually done as part of the `requestPermissionsImpl()` method in Objective-C

cc @Kavantix in case this was done of a PR you did to update @ened's fork and may have an answer

Reply:
Now that you mention it I did also wonder the same thing when I was updating the PR.
This should probably be refactored such that neither method calls the result.
It should just be called in initialize

Reply:
The reason it's called in `requestPermissionsImpl()` is because it is meant to return the result of answering the permissions prompt on iOS/macOS

Reply:
This was adjusted in the latest MR. Please re-open a comment when needed.

Reply:
@ened to clarify, did you mean there was a comment that I've missed I could've been reopened and replied to or to re-open this one if there's something to follow-up on?

Reply:
The latter one, please reopen if further changes are needed. :-)